app/uk/gov/hmrc/plasticpackagingtax/registration/forms/contact/AddressMapper.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.forms.contact
+
+import play.api.data.Forms.{mapping, optional}
+import play.api.data.{Form, Forms}
+
+trait AddressMapper {
+
+  val formToModel
+    : (String, Option[String], Option[String], String, Option[String], String) => Address = (
+    addressLine1: String,
+    addressLine2: Option[String],
+    addressLine3: Option[String],
+    townOrCity: String,
+    postCode: Option[String],
+    countryCode: String
+  ) => new Address(addressLine1, addressLine2, addressLine3, townOrCity, postCode, countryCode)
+
+  val modelToForm
+    : Address => Some[(String, Option[String], Option[String], String, Option[String], String)] =
+    (address: Address) =>
+      Some(
+        (address.addressLine1,
+         address.addressLine2,
+         address.addressLine3,
+         address.townOrCity,
+         address.postCode,
+         address.countryCode
+        )
+      )
+
+  // Use this to unbind the form data from the request to get around the problem of conditional mandatoryIfEqual
+  // mapping not binding postcode
+  val dataExtractor: Form[Address] =
```

Comment:
I missed/forgot about this "fix" for the conditional mapping because the extractor was in a separate class.  Maybe it would be better if this method was in the "Address" object (with a note explaining why it is needed)?

Reply:
Agreed